### ISSUE TITLE
Question of optimizer updates

### ISSUE THREAD

**Issue by takisu0916** [Fictionarry/DNGaussian#38]
Very interesting work！ I have a question: In the official 3Dgs 'train.py', the ’gaussians.optimizer.step()’ is executed only once per iteration, but in your code, ‘gaussians.optimizer.step()’ is executed four times for different modules. Is this fair?

**Comment by Fictionarry** [comment]
Hi, actually there are three update operations, of which the first two would not influence other attributes (like color or shape, or influence the densification) but only Gaussian centers and opacities respectively through our two proposed depth regularizations. This can be regarded approximately as adding some regularization terms to the total loss and updating one time. We separate them for convenience.

Besides, I guess you are worrying about whether 3DGS can become better when being trained by more iterations. Indeed it would not. In sparse-view settings, using more iterations would significantly aggravate overfitting for the original 3DGS. The core problem is how to escape the model from overfitting, rather than underfitting.

**Comment by takisu0916** [comment]
If we're discussing gradient accumulation, it would make more sense to use multiple backward() calls within a single iteration for different sub-losses. However, your use of multiple step() calls is puzzling. Have you conducted any ablation experiments to demonstrate the difference between performing step() for each sub-loss separately versus accumulating the total_loss and then performing a single step()?

**Comment by Fictionarry** [comment]
Well, to be honest, I don't understand what you are worried about... If you really care weather the ablation between our method and original 3DGS is fair, I have already explained it. First, the two additional updates belong to our contribution. If you have already read the paper, you can find that the additional updates correspond to two depth-regularization losses that do not exist in the original 3DGS, which are both irrelevant to the optimization process in original 3DGS. So, the two updates would not bring extra contributions except depth regularization. Second, original 3DGS would not benefit from more updates. You can just try by yourself to verify if this is the truth. 

Then, if this is fair, I can do everything I like to design my model. In fact, the main reason is that I think adding all the losses together and updating once is very ugly. They should naturally belong to different stages. I do not think the multiple updates are significantly different from only one in this work, considering they correspond to different types of losses and gradient flows. I just like such a process. It's irrelevant to gradient accumulation.

**Comment by Fictionarry** [comment]
Oh, I forgot one thing. Separating depth regularization and color supervision can avoid the influence of depth losses on the densification process. I do not want them to influence the original optimization process, otherwise, I'm afraid there might be some unnecessary troubles.

**Comment by takisu0916** [comment]
I don't have any other questions. Thank you for your patient response ！
